### ISSUE TITLE
Incorrect inheritance with forward declared templated classes

### ISSUE THREAD

**Issue by jureslak** [doxygen/doxygen#8177]
Doxygen does not correctly recognize inheritance of templated classes, if a derived class has been forward declared.

Generating documentation for a header
```cpp
template <typename T>
class B;

 /// Class A that takes care of A.
template <typename T>
class A {
  public:
    ///  Does public stuff.
    virtual int doStuff(int a) { return 8; }
};

/// Class B that takes case of B.
template <typename T>
class B : public A {
  public:
    virtual int doStuff(int a) { return 9; }
};

```
produces a warning 
```
/.../a.hpp:22: warning: Member doStuff(int a) (function) of class B is not documented.
```
and does not identify `B` as a derived class of `A`.

**Expected behavior**
`B` should be a marked as a subclass of `A` and the docs should be inherited.

**To Reproduce**
Generate a default Doxyfile in the same directory as the code above, and run the `doxygen` command. 
The example works ok without the forward declaration of `B` (which is superfluous in this case, but is sometimes required). 

Files (remove the `.txt`)
[Doxyfile.txt](https://github.com/doxygen/doxygen/files/5549071/Doxyfile.txt)
[a.hpp.txt](https://github.com/doxygen/doxygen/files/5552657/a.hpp.txt)

**Version**
Doxygen versions from 1.8.13 to 1.8.20 were tested (arch linux), and the problem first appears in 1.8.17 and is still present in the current version (1.8.20).

**Additional context**
This has already been mentioned in #7897 but I believe this issue make it more obvious what the actual problem is and easier to discover by search engines. Feel free to close it as a duplicate though.

A possible workaround is to define a preprocessor symbol in the `PREDEFINED` config option in the `Doxyfile`, e.g.
```
PREDEFINED = DOXYGEN_RUN
```
and wrap the forward declaration(s) in `#ifndef DOXYGEN_RUN / #endif` pair, so that the compiler sees the definition, but `doxygen` doesn't.

**Comment by albert-github** [comment]
I checked the different commits after the 1.8.16 release and the first commit where the warning occurs en the problem occurs is (d2b8ae16cb99621827f5dd860222c27e629f5a7f ):
```
Commit: d2b8ae16cb99621827f5dd860222c27e629f5a7f [d2b8ae1]
Date: Monday, October 7, 2019 9:02:04 PM
Merge branch 'std'
```
i.e. for
```
Use smartpointers to manage the lifetime of Entry objects
```
The commit before this commit (4e21d19cf63b17aadb62905535262deb208a0c73 ):
```
Commit: 4e21d19cf63b17aadb62905535262deb208a0c73 [4e21d19]
Date: Saturday, September 28, 2019 12:08:11 PM
Merge pull request #7282 from albert-github/feature/issue_7274

issue #7274 Subpages no longer possible under mainpage
```
is still OK.

**Comment by doxygen** [comment]
Please verify if the commit fixes the problem.

**Comment by jureslak** [comment]
I compiled doxygen with the lastest master (fa65bb3) and the issue is fixed. The docs for `B` generate as expected now:
![image](https://user-images.githubusercontent.com/1394590/99512141-6d8abf80-2989-11eb-8485-5a985967e79f.png)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
